.dev/gather_models.py

```diff
@@ -50,25 +50,21 @@ def get_final_iter(config):
 
 
 def get_final_results(log_json_path, iter_num):
-    result_dict = dict()
+    result_dict = {}
     last_iter = 0
     with open(log_json_path, 'r') as f:
-        for line in f.readlines():
+        for line in f:
             log_line = json.loads(line)
             if 'mode' not in log_line.keys():
                 continue
-
-            # When evaluation, the 'iter' of new log json is the evaluation
-            # steps on single gpu.
-            flag1 = ('aAcc' in log_line) or (log_line['mode'] == 'val')
-            flag2 = (last_iter == iter_num - 50) or (last_iter == iter_num)
+            flag1 = 'aAcc' in log_line or log_line['mode'] == 'val'
+            flag2 = last_iter in [iter_num - 50, iter_num]
             if flag1 and flag2:
-                result_dict.update({
+                result_dict |= {
                     key: log_line[key]
                     for key in RESULTS_LUT if key in log_line
-                })
+                }
```

Comment:
`|=` might unsupport dict

Reply:
That's supported, but you're right. It's not compatible with `Python < 3.9`. I return some changes to the previous version.